tests/test_adapter.py

```diff
@@ -2008,3 +2012,99 @@ def test_exception_handling(self, mock_stage, mock_download_subset,
 
         mock_stage.assert_not_called()
         mock_rmtree.assert_called_once_with(self.tmp_dir)
+
+    @patch('hoss.dimension_utilities.get_fill_slice')
+    @patch('hoss.utilities.uuid4')
+    @patch('hoss.adapter.mkdtemp')
+    @patch('shutil.rmtree')
+    @patch('hoss.bbox_utilities.download')
+    @patch('hoss.utilities.util_download')
+    @patch('hoss.adapter.stage')
+    def test_edge_aligned_no_bounds_end_to_end(self, mock_stage, mock_util_download,
+                         mock_geojson_download, mock_rmtree,
+                         mock_mkdtemp, mock_uuid,
+                         mock_get_fill_slice):
+        """ Ensure a request for a collection that contains dimension variables
+            with edge-aligned grid cells is correctly processed regardless of
+            whether or not a bounds variable associated with that dimension
+            variable exists.
+
+        """
+        expected_output_basename = 'opendap_url_global_asr_obs_grid_subsetted.nc4'
+        expected_staged_url = f'{self.staging_location}{expected_output_basename}'
+
+        mock_uuid.side_effect = [Mock(hex='uuid'), Mock(hex='uuid2')]
+        mock_mkdtemp.return_value = self.tmp_dir
+        mock_stage.return_value = expected_staged_url
+
+        dmr_path = write_dmr(self.tmp_dir, self.atl16_dmr)
+
+        dimensions_path = f'{self.tmp_dir}/dimensions.nc4'
+        copy('tests/data/ATL16_prefetch.nc4', dimensions_path)
+
+        all_variables_path = f'{self.tmp_dir}/variables.nc4'
+        copy('tests/data/ATL16_variables.nc4', all_variables_path)
+
+        mock_util_download.side_effect = [dmr_path, dimensions_path,
+                                          all_variables_path]
+
+        message = Message({
+            'accessToken': 'fake-token',
+                'callback': 'https://example.com/',
+                'sources': [{
+                    'collection': 'C1238589498-EEDTEST',
+                    'shortName': 'ATL16',
+                    'variables': [{'id': '',
+                                'name': self.atl16_variable,
+                                'fullPath': self.atl16_variable}]}],
+                'stagingLocation': self.staging_location,
+                'subset': {'bbox': [77, 71.5, 88, 74.75]},
+                'user': 'jlovell',
+        })
+
+        hoss = HossAdapter(message, config=config(False), catalog=self.input_stac)
```

Comment:
Nitpick question: Does `config=config(False)` just mean that we're not using a configuration file to create this? If so, is it necessary to have this since the default behavior assumes there's no configuration file if one isn't included?

Reply:
Ah, so the configuration can be drawn from the environment and not necessarily an input config file? (Or something along those lines)

Reply:
Oh got it. I've mixed up these two config object types before.